File: rust-rocket-sdk/src/catalog.rs
//! Shared workshop keys. Keep the flag recipe aligned with FLAG_TEMPLATE.md.
pub const FLAGS: [&str; 16] = [
    "new-dashboard",
    "api-v2",
    "enhanced-submit",
    "ExpressCheckout",
    "beta-access",
    "filter-always-on",
    "filter-percentage",
    "filter-targeting",
    "filter-user-claims",
    "filter-time-window",
    "filter-country",
    "filter-browser-family",
    "filter-browser-language",
    "filter-device-type",
    "filter-os",
    "filter-context-property",
];

pub const FILTERS: [(&str, &str, &str); 11] = [
    ("filter-always-on", "AlwaysOn", "On for both presets."),
    (
        "filter-percentage",
        "Percentage 50%",
        "Sticky by identity; neither preset prescribes an outcome.",
    ),
    (
        "filter-targeting",
        "Targeting users=alice",
        "Matching on; Non-matching off.",
    ),
    (
        "filter-user-claims",
        "UserClaims role=admin",
        "Matching on; Non-matching off.",
    ),
    (
        "filter-time-window",
        "TimeWindow 2020–2099",
        "On for both presets while the window is open.",
    ),
    (
        "filter-country",
        "Country US",
        "Matching on; Non-matching off.",
    ),
    (
        "filter-browser-family",
        "BrowserFamily Chrome",
        "Matching on; Non-matching off.",
    ),
    (
        "filter-browser-language",
        "BrowserLanguage en",
        "Matching on; Non-matching off.",
    ),
    (
        "filter-device-type",
        "DeviceType Macintosh",
        "Unsupported desktop detection in published 0.4.0: native false even for Matching. Recipe is unchanged.",
    ),
    (
        "filter-os",
        "OperatingSystem Mac",
        "Matching on; Non-matching off.",
    ),
    (
        "filter-context-property",
        "ContextProperty Order.Vip=true",
        "Matching on; Non-matching off.",
    ),
];

File: astro-sdk/src/sample/catalog.ts
/** Shared demo names from docs/FLAG_TEMPLATE.md — keep keys identical to the dashboard. */

export const demoKeys = [
  'new-dashboard',
  'api-v2',
  'enhanced-submit',
  'ExpressCheckout',
  'beta-access',
] as const;

export const filterCatalog = [
  { key: 'filter-always-on', name: 'AlwaysOn', rule: 'Always on in both presets' },
  {
    key: 'filter-percentage',
    name: 'Percentage',
    rule: '50%; sticky by identity, not a guaranteed match',
  },
  { key: 'filter-targeting', name: 'Targeting', rule: 'User alice' },
  { key: 'filter-user-claims', name: 'UserClaims', rule: 'role = admin' },
  {
    key: 'filter-time-window',
    name: 'TimeWindow',
    rule: 'Open from 2020 through 2099',
  },
  { key: 'filter-country', name: 'Country', rule: 'US' },
  { key: 'filter-browser-family', name: 'BrowserFamily', rule: 'Chrome' },
  { key: 'filter-browser-language', name: 'BrowserLanguage', rule: 'en' },
  { key: 'filter-device-type', name: 'DeviceType', rule: 'Macintosh' },
  { key: 'filter-os', name: 'OperatingSystem', rule: 'Mac' },
  {
    key: 'filter-context-property',
    name: 'ContextProperty',
    rule: 'Order.Vip = true',
  },
] as const;

export const filterKeys = filterCatalog.map((f) => f.key);

/**
 * HTTP-derived filters need request metadata (country, UA, Accept-Language).
 * Published Astro server buildEvalContext only supplies identity, groups,
 * claims, and entity — so these rows stay honest unsupported for forged
 * query-preset evaluation. Live edge workers can still enforce them.
 */
export const httpDerivedFilterKeys = new Set([
  'filter-country',
  'filter-browser-family',
  'filter-browser-language',
  'filter-device-type',
  'filter-os',
]);

export const orders = {
  vip: { Id: 'ord-vip', Vip: true, Total: 240 },
  standard: { Id: 'ord-standard', Vip: false, Total: 60 },
} as const;

export type PresetName = 'matching' | 'nonmatching';

export const presets: Record<
  PresetName,
  {
    identity: string;
    claims: { role: string };
    order: 'vip' | 'standard';
    country: string;
    language: string;
    userAgent: string;
  }
> = {
  matching: {
    identity: 'alice',
    claims: { role: 'admin' },
    order: 'vip',
    country: 'US',
    language: 'en-US,en;q=0.9',
    userAgent:
      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
  },
  nonmatching: {
    identity: 'bob',
    claims: { role: 'user' },
    order: 'standard',
    country: 'CA',
    language: 'fr-FR,fr;q=0.9',
    userAgent:
      'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0',
  },
};
